ci/run-ci.py

```diff
@@ -0,0 +1,475 @@
+#   We ensure the GKE cluster always gets destroyed at the end of the execution
+#   of this script.
+#
+#   We use 2 mechanisms:
+#   - When the script exits, we destroy the cluster as the final step. That's
+#     the case even if the script exits with an error or with an interruption
+#     (Ctrl+C). See `trap` command.
+#   - In any other "abrupt shutdown" case (system shutdown, Travis build
+#     canceled, etc), we use a daily scheduled task to destroy stale clusters
+#     after 24h. See https://console.cloud.google.com/functions/details/us-central1/clean-substra-tests-ci-deployment
+#
+# USEFUL LINKS
+#   Travis build logs: https://travis-ci.org/github/SubstraFoundation/substra-tests
+#   Stale cluster deletion script:
+#       https://console.cloud.google.com/functions/details/us-central1/clean-substra-tests-ci-deployment
+import os
+import time
+import json
+import shutil
+import string
+import random
+import argparse
+import functools
+import subprocess
+import sys
+
+CLUSTER_NAME_ALLOWED_PREFIX = 'substra-tests'
+CLUSTER_NAME = ''
+CLUSTER_MACHINE_TYPE = 'n1-standard-8'
+
+CLUSTER_VERSION = '1.15.11-gke.1'
+CLUSTER_PROJECT = 'substra-208412'
+CLUSTER_ZONE = 'europe-west4-a'
+
+SERVICE_ACCOUNT = 'substra-tests@substra-208412.iam.gserviceaccount.com'
+KEY_SERVICE_ACCOUNT = 'substra-208412-3be0df12d87a.json'
+
+SUBSTRA_TESTS_BRANCH = 'nightly-tests'
+SUBSTRA_BACKEND_BRANCH = 'master'
+HLF_K8S_BRANCH = 'master'
+
+DIR = os.path.dirname(os.path.realpath(__file__))
+CHARTS_DIR = os.path.realpath(os.path.join(DIR, '../charts/'))
+KEYS_DIR = os.path.realpath(os.path.join(os.getenv('HOME'), '.local/'))
+SOURCE_DIR = os.path.realpath(os.path.join(DIR, 'src'))
+
+KUBE_CONTEXT = ''
+RUN_TAG = ''.join(random.choice(string.ascii_letters + '0123456789') for _ in range(10))
+
+KANIKO_CACHE_TTL = '168h'  # 1 week
+
+
+def call(cmd):
+    print(f'+ {cmd}')
+    return subprocess.check_call([cmd], shell=True)
+
+
+def call_output(cmd, print_cmd=True):
+    if print_cmd:
+        print(f'+ {cmd}')
+    return subprocess.check_output([cmd], shell=True, stderr=subprocess.STDOUT).decode().strip()
+
+
+def cluster_name(value):
+    """
+    Validate the --cluster-name argument
+    The cluster name must start with 'substra-tests'.
+    This is to ensure the cluster gets picked up by the stale cluster deletion script.
+    """
+
+    if not value.startswith(CLUSTER_NAME_ALLOWED_PREFIX):
+        raise argparse.ArgumentTypeError(
+            f'Invalid cluster name "{value}". '
+            f'The cluster name must start with "{CLUSTER_NAME_ALLOWED_PREFIX}".')
+
+    if len(value) > 35:
+        raise argparse.ArgumentTypeError(
+            f'Invalid cluster name "{value}". '
+            f'The cluster name must not be longer than 35 characters.')
+
+    return value
+
+
+def arg_parse():
+
+    global KEYS_DIR
+    global CLUSTER_NAME
+    global SUBSTRA_TESTS_BRANCH
+    global SUBSTRA_BACKEND_BRANCH
+    global HLF_K8S_BRANCH
+    global KANIKO_CACHE_TTL
```

Comment:
I agree in principle, but wouldn't the constraints introduced by doing that outweight the benefits?

Reply:
Yes, same as Auélien.
We could use only a mutable structure as dictionnary  and pass it as reference in each call.
But we should remember that's a script to launch the test and it's easier to have something simple.

Reply:
I do agree, it's debatable. In my opinion, it would make the code simpler to use a mutable structure.

As said by @Kelvin-M and as said in my main comment, it's a script to launch the tests, so it's good for me.